src/utils/index.ts

```diff
@@ -112,7 +112,7 @@ export const reparentActor = (
     parent: Clutter.Actor | null,
     first = false
 ) => {
-    if (!actor || !parent) return;
+    if (!actor || !parent || actor.get_parent() == parent) return;
```

Comment:
Reuse the `currentParent` variable to avoid getting the parent twice (avoids a js -> c call)